zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/cacheDatabase.kt

```diff
@@ -0,0 +1,35 @@
+package app.cash.zipline.loader
+
+import com.squareup.sqldelight.EnumColumnAdapter
+import com.squareup.sqldelight.db.SqlDriver
+
+// in src/commonMain/kotlin
+expect class DriverFactory {
+  fun createDriver(): SqlDriver
+}
+
+// TODO make internal / upstream to SqlDelight
+expect class SQLiteException : SQLException
+expect open class SQLException : Exception
+
+fun createDatabase(driverFactory: DriverFactory): Database {
+  val driver = driverFactory.createDriver()
+  val database = Database(
+    driver,
+    filesAdapter = Files.Adapter(
+      file_stateAdapter = EnumColumnAdapter()
+    )
+  )
+  return database
+}
+
+fun createDatabase(driver: SqlDriver): Database {
+  val database = Database(
+    driver,
+    filesAdapter = Files.Adapter(
+      file_stateAdapter = EnumColumnAdapter()
+    )
+  )
+  Database.Schema.create(driver)
```

Comment:
The JVM implementation of `DriverFactory` already calls this so I would expect this one to fail since the tables already exist.

Reply:
It doesn't seem to from initial early testing but I didn't thoroughly check